src/open_inwoner/laposta/client.py

```diff
@@ -71,6 +88,26 @@ def remove_subscription(self, list_id: str, member_id: str) -> Member | None:
 
         return Member(**data["member"])
 
+    @cache_result(
+        "laposta_list_subscriptions:{email}", timeout=settings.CACHE_LAPOSTA_API_TIMEOUT
+    )
+    def get_subscriptions_for_email(
+        self, list_ids: list[str], email: str
+    ) -> dict[str, str]:
+        # The API cannot deal with urlencoded email addresses that contain a `+`,
+        # so instead we fetch all members for a given list and check if there is a
+        # member with a matching email
+        subscribed_to = {}
+        for list_id in list_ids:
+            response = self.get("member", params={"list_id": list_id})
+            if response.status_code == 200:
+                data = get_json_response(response)
+                for member_data in data["data"]:
+                    member = Member(**member_data["member"])
```

Comment:
The list of all members could be very long: maybe we don't need throwaway instances here but can use the dict.

Reply:
The `member_id` is not the same for the same email across different `list_id`s unfortunately, so I would still have to parse the `member_id`